### PR TITLE
[WFLY-12317][JBEAP-17293] changing of txn subsystem node identifier requires JVM restart

### PR DESCRIPTION
**Pull request by ochaloup**
https://issues.jboss.org/browse/WFLY-12317
https://issues.jboss.org/browse/JBEAP-17293

### CODE REVIEW

**Comment by kabir** [comment]
@ochaloup I think this needs a model version bump since you are changing the model (although in a small way)

**Review by kabir — CHANGES_REQUESTED** [review]
We agreed in chat a version bump is needed

**Comment by ochaloup** [comment]
@kabir hi. I processed of the version upgrade. But I'm not sure if I did not miss something necessary for such change. I'm sorry for my ignorance in this space but I would like ask you for hint what should be added, adjusted and in case if a test needs to be added.

File: transactions/src/main/java/org/jboss/as/txn/subsystem/Namespace.java [hunk 1]
```diff
@@ -44,12 +44,13 @@
     TRANSACTIONS_3_0("urn:jboss:domain:transactions:3.0"),
     TRANSACTIONS_4_0("urn:jboss:domain:transactions:4.0"),
     TRANSACTIONS_5_0("urn:jboss:domain:transactions:5.0"),
+    TRANSACTIONS_5_1("urn:jboss:domain:transactions:5.1"),
```
Comment (kabir) [inline transactions/src/main/java/org/jboss/as/txn/subsystem/Namespace.java]: As the xml has not changed, you don't need a new xsd (it is ok for the xsd and model versions to diverge, we have other subsystems doing that)

File: transactions/src/main/java/org/jboss/as/txn/subsystem/Namespace.java [hunk 2]
```diff
@@ -44,12 +44,13 @@
...
     ;
 
     /**
      * The current namespace version.
      */
-    public static final Namespace CURRENT = TRANSACTIONS_5_0;
+    public static final Namespace CURRENT = TRANSACTIONS_5_1;
```
Comment (kabir) [inline transactions/src/main/java/org/jboss/as/txn/subsystem/Namespace.java]: Not needed

File: transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionExtension.java [hunk 3]
```diff
@@ -134,5 +134,6 @@ public void initializeParsers(ExtensionParsingContext context) {
         context.setSubsystemXmlMapping(SUBSYSTEM_NAME, Namespace.TRANSACTIONS_3_0.getUriString(), TransactionSubsystem30Parser::new);
         context.setSubsystemXmlMapping(SUBSYSTEM_NAME, Namespace.TRANSACTIONS_4_0.getUriString(), TransactionSubsystem40Parser::new);
         context.setSubsystemXmlMapping(SUBSYSTEM_NAME, Namespace.TRANSACTIONS_5_0.getUriString(), TransactionSubsystem50Parser::new);
+        context.setSubsystemXmlMapping(SUBSYSTEM_NAME, Namespace.TRANSACTIONS_5_1.getUriString(), TransactionSubsystem51Parser::new);
```
Comment (kabir) [inline transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionExtension.java]: Not needed

File: transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystem51Parser.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ */
+package org.jboss.as.txn.subsystem;
+
+/**
+ * The {@link org.jboss.staxmapper.XMLElementReader} that handles the version 5.1 of Transaction subsystem xml.
+ *
+ * The update contains only requirement of JVM restart for change on 'node-id'.
+ */
+class TransactionSubsystem51Parser extends TransactionSubsystem50Parser {
```
Comment (kabir) [inline transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystem51Parser.java]: Delete this class

File: transactions/src/main/resources/schema/wildfly-txn_5_1.xsd [hunk 5]
```diff
@@ -0,0 +1,339 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
```
Comment (kabir) [inline transactions/src/main/resources/schema/wildfly-txn_5_1.xsd]: Delete this file

File: transactions/src/main/resources/subsystem-templates/transactions.xml [hunk 6]
```diff
@@ -2,7 +2,7 @@
 <!--  See src/resources/configuration/ReadMe.txt for how the configuration assembly works -->
 <config>
    <extension-module>org.jboss.as.transactions</extension-module>
-   <subsystem xmlns="urn:jboss:domain:transactions:5.0">
+   <subsystem xmlns="urn:jboss:domain:transactions:5.1">
```
Comment (kabir) [inline transactions/src/main/resources/subsystem-templates/transactions.xml]: This change is not needed

File: transactions/src/test/resources/org/jboss/as/txn/subsystem/subsystem.xml [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<subsystem xmlns="urn:jboss:domain:transactions:5.0">
+<subsystem xmlns="urn:jboss:domain:transactions:5.1">
```
Comment (kabir) [inline transactions/src/test/resources/org/jboss/as/txn/subsystem/subsystem.xml]: Change not needed

**Review by kabir — CHANGES_REQUESTED** [review]
@ochaloup You did too much :)

Please see the individual comments for guidance. If you have any problems with getting the transformers tests to pass with my suggestions, ping me

File: transactions/src/main/resources/schema/wildfly-txn_5_1.xsd (hunk 5 above)
Comment (ochaloup) [reply transactions/src/main/resources/schema/wildfly-txn_5_1.xsd]: @kabir I really don't need this file when I do define a new schema version? I wonder first as I expected this one will be necessary and as a second point I fail with test `AbstractSubsystemBaseTest#testSchema` (I'm with investigation on it)

Comment (kabir) [reply transactions/src/main/resources/schema/wildfly-txn_5_1.xsd]: @ochaloup You do not need a new schema version. Nothing has changed in the xml parsing, only in the management model. Sorry if I was not clear

Comment (ochaloup) [reply transactions/src/main/resources/schema/wildfly-txn_5_1.xsd]: @kabir thanks Kabir. It's rather my trouble as I'm still not well familiar with the dependencies needed for upgrade.

**Comment by ochaloup** [comment]
@kabir I made changes based on your guidelines. Thanks! If you can be so kind please recheck the PR once again :)

File: transactions/src/main/java/org/jboss/as/txn/subsystem/Namespace.java (hunk 1 above)
Comment (kabir) [reply transactions/src/main/java/org/jboss/as/txn/subsystem/Namespace.java]: @ochaloup I think it looks fine now apart from this namespace is not needed

Comment (ochaloup) [reply transactions/src/main/java/org/jboss/as/txn/subsystem/Namespace.java]: fixed now

**Comment by ochaloup** [comment]
@mjurc @msve can you please confirm this PR can be merged without RFE?

**Comment by ochaloup** [comment]
@kabir what do you think about the current state of the PR? Could be merged?

(I removed the comment as I agreed with QE @istraka he is fine to get this being processed. The comment which removed belonged to a different issue - my erro)

**Comment by ochaloup** [comment]
@kabir just for you info there is currently conversation at https://issues.jboss.org/browse/WFLY-12317

**Pull request merged**